### PR TITLE
Reduce carbon footprint with better error messages

### PR DESCRIPTION
**Pull request by daysleeperz**
By decreasing the amount of time developers need to debug non descriptive error messages we can drastically reduce electricity wasted.

From phone, refactor as you like. Take it from here bye

### CODE REVIEW

**Comment by daysleeperz** [comment]
@dotnet-policy-service agree

File: src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SqlBuffer.cs [hunk 1]
```diff
@@ -249,12 +249,16 @@ internal decimal Decimal
                                 }
                                 catch (OverflowException)
                                 {
-                                    throw new OverflowException(SQLResource.ConversionOverflowMessage);
+                                    throw new OverflowException(SQLResource.ConversionOverflowMessage + " for " + sqlValue.ToString());
                                 }
                             }
                         }
-                        throw new OverflowException(SQLResource.ConversionOverflowMessage);
+                        throw new OverflowException(SQLResource.ConversionOverflowMessage+ " for " + sqlValue.ToString());
```
Comment (Wraith2) [inline src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SqlBuffer.cs:256]: The sqlValue should be passed as a parameter to a function which uses a fomatter to correctly display the message in the user's langauge. Have a look at other similar exception messages that have parameter data in the project.

File: src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SqlBuffer.cs [hunk 2]
```diff
@@ -249,12 +249,16 @@ internal decimal Decimal
...
                     }
+
+SqlDecimal sqlValue = new(_value._numericInfo._precision, _value._numericInfo._scale, _value._numericInfo._positive,
```
Comment (Wraith2) [inline src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SqlBuffer.cs:259]: What is this for? It doesn't seem to be used.

**Comment by JRahnama** [comment]
Closing this due to inactivity. Feel free to open a new PR.

**Pull request closed without merging**